CMakeLists.txt

```diff
@@ -115,12 +123,21 @@ message(STATUS "HIP runtime:      ${HIP_RUNTIME}")
 if(NOT "${HIP_COMPILER}" MATCHES "clang")
   message(FATAL_ERROR "RCCL requires clang-based compiler (hipcc)")
 endif()
-find_program(hipcc_executable hipcc)
-message(STATUS "hipcc executable: ${hipcc_executable}")
-execute_process(
-  COMMAND         bash "-c" "${hipcc_executable} --version | grep 'HIP version' | awk -F\" \" '{ printf $3}' | awk -F\"-\" '{ printf $1}'"
-  OUTPUT_VARIABLE hipcc_version_string)
-message(STATUS "hipcc version:    ${hipcc_version_string}")
+if("${CMAKE_CXX_COMPILER}" MATCHES ".*hipcc$")
+  find_program(hipcc_executable hipcc)
+  message(STATUS "hipcc executable: ${hipcc_executable}")
+  execute_process(
+    COMMAND         bash "-c" "${hipcc_executable} --version | grep 'HIP version' | awk -F\" \" '{ printf $3}' | awk -F\"-\" '{ printf $1}'"
+    OUTPUT_VARIABLE hipcc_version_string)
+  message(STATUS "hipcc version:    ${hipcc_version_string}")
+elseif("${CMAKE_CXX_COMPILER}" MATCHES ".*amdclang")
+  find_program(hipcc_executable amdclang)
+  message(STATUS "amdclang executable: ${hipcc_executable}")
+  execute_process(
+    COMMAND         bash "-c" "${hipcc_executable} --version | grep 'AMD clang version' | awk -F\" \" '{ printf $4}'"
+    OUTPUT_VARIABLE hipcc_version_string)
+  message(STATUS "amdclang version:    ${hipcc_version_string}")
+endif()
```

Comment:
Not a functional change, but it would aid comprehension if we renamed all the `hipcc_` variables to something more generic.

Reply:
Did we add CI testing for this option?

Reply:
We might want to reduce the amount of duplicate code here. The first if/else statement should be enough to gather info to generalize the second if/else block.

Reply:
> Did we add CI testing for this option?
no... waiting on info about ROCm version which enforces switch from hipcc to amdclang++

Reply:
> We might want to reduce the amount of duplicate code here. The first if/else statement should be enough to gather info to generalize the second if/else block.

may add an else block for exit if the compiler is neither hipcc nor amdclang

Reply:
> > We might want to reduce the amount of duplicate code here. The first if/else statement should be enough to gather info to generalize the second if/else block.
> 
> may add an else block for exit if the compiler is neither hipcc nor amdclang

I don't think you need an if/else block here at all. Once you have the required info from the first if/else statement in this PR, this one can just be generalized. When cmake tries to find a package and unsuccessful, it will fail anyway.

Reply:
reduced if/else blocks